src/sqlfluff/core/templaters/jinja.py

```diff
@@ -383,28 +403,84 @@ def process(
 
         undefined_variables = set()
 
-        class Undefined:
-            """Similar to jinja2.StrictUndefined, but remembers, not fails."""
+        if "templating" not in config.get("ignore"):
```

Comment:
I suggest reviewing this section with "Hide whitespace" enabled. We retain the old version of `class Undefined`, and we add a new implementation for the `ignore=templating` case.